lib/Utils/AffineMapUtils.cpp

```diff
@@ -0,0 +1,42 @@
+#include "lib/Utils/AffineMapUtils.h"
+
+#include "mlir/include/mlir/IR/AffineMap.h"          // from @llvm-project
+#include "mlir/include/mlir/IR/Builders.h"           // from @llvm-project
+#include "mlir/include/mlir/IR/BuiltinAttributes.h"  // from @llvm-project
+#include "mlir/include/mlir/Support/LLVM.h"          // from @llvm-project
+
+namespace mlir {
+namespace heir {
+
+LogicalResult makeExplicit1DMapping(AffineMap map, unsigned rank,
+                                    SmallVector<int64_t> &result) {
+  if (map.getNumResults() != 1) return failure();
+  if (map.getNumDims() != 1) return failure();
+  if (map.getNumSymbols() != 0) return failure();
+
+  OpBuilder b(map.getContext());
+  result.resize(rank);
+  for (size_t permInput = 0; permInput < rank; ++permInput) {
+    SmallVector<Attribute> operandConstants{b.getIndexAttr(permInput)};
+    SmallVector<Attribute> results;
+    if (failed(map.constantFold(operandConstants, results))) {
+      assert(false && "constant folding should never fail here");
+      return failure();
+    }
```

Comment:
so each loop iteration is calculating the constantFold at operandConstant? can you assemble the a whole list of the permInput index attrs and feed that to `constantFold` and then the result vector is exactly the result (assuming you `map_vector` from attribute to int64_t after)

Reply:
`constantFold` here is just evaluating the affine map at a given input. I changed it to be more functional, but I'm not convinced it's significantly better.